[Technical - LDAP] error when user have no email field in LDAP

### Problem Statement

<!-- For Bug: What happened? Where? Who is impacted? Environment (browser, OS, device)? -->
<!-- For Feature: What user need or goal are we addressing? Why now? -->
<!-- For Task: What is the objective, scope, and success signal? -->

### Bug

In some case, when we use ldap we can have a NPE.
When we read email attribute, we do attr.getAttribute("email"), and then if not null, we do attr.getAttribute("email").getValue.toString()So, when user object in ldap have not the attribute at all, attr.getAttribute is null, and the check is correct => no NPEBut in some case, the attr.getAttribute is not null, but the attr.getAttribute.getValue is null, and then it generates an NPE :

> Caused by: java.lang.NullPointerException: Cannot invoke "Object.toString()" because the return value of "org.picketlink.idm.api.Attribute.getValue()" is null at 
> org.exoplatform.services.organization.idm.EntityMapperUtils.populateUser(EntityMapperUtils.java:94) at 
> org.exoplatform.services.organization.idm.EntityMapperUtils.populateUser(EntityMapperUtils.java:63) at 
> org.exoplatform.services.organization.idm.UserDAOImpl.populateUser(UserDAOImpl.java:970) at 
> org.exoplatform.services.organization.idm.UserDAOImpl.getPopulatedUser(UserDAOImpl.java:942) at 
> org.exoplatform.services.organization.idm.UserDAOImpl.findUserByName(UserDAOImpl.java:361) at 
> org.exoplatform.services.organization.idm.cache.CacheableUserHandlerImpl.findUserByName(CacheableUserHandlerImpl.java:169) at 
> org.exoplatform.services.organization.idm.externalstore.PicketLinkIDMExternalStoreService.lambda$getEntity$3(PicketLinkIDMExternalStoreService.java:373) at 
> org.exoplatform.services.organization.idm.externalstore.PicketLinkIDMExternalStoreService.lambda$new$0(PicketLinkIDMExternalStoreService.java:104)

We need to improve the test to prevent this error

### Feature — Solution & Expected Benefits (only if Type = Feature)

#### Proposed Solution
-

#### Expected Benefits
<!--
- Improves ...
- Reduces ...
- Measurable outcome: ...
-->

### Additional Context

MEED-10404

--- a/component/identity/src/main/java/org/exoplatform/services/organization/idm/EntityMapperUtils.java
+++ b/component/identity/src/main/java/org/exoplatform/services/organization/idm/EntityMapperUtils.java
@@ -72,7 +72,7 @@ public class EntityMapperUtils {
       throw new IllegalStateException("Attributes map is mandatory");
     }
     boolean changed = false;
-    if (attrs.containsKey(USER_CREATED_DATE)) {
+    if (attrs.containsKey(USER_CREATED_DATE) && attrs.get(USER_CREATED_DATE) != null && attrs.get(USER_CREATED_DATE).getValue() != null) {
       Date origCreatedDate = user.getCreatedDate();
       Date createdDate = null;
       try {
@@ -90,12 +90,12 @@ public class EntityMapperUtils {
       }
       changed |= checkIfChanged && !Objects.equals(createdDate, origCreatedDate);
     }
-    if (attrs.containsKey(USER_EMAIL)) {
+    if (attrs.containsKey(USER_EMAIL) && attrs.get(USER_EMAIL) != null && attrs.get(USER_EMAIL).getValue() != null) {
       String email = attrs.get(USER_EMAIL).getValue().toString();
       changed |= checkIfChanged && !Objects.equals(email, user.getEmail());
       user.setEmail(email);
     }
-    if (attrs.containsKey(USER_FIRST_NAME)) {
+    if (attrs.containsKey(USER_FIRST_NAME) && attrs.get(USER_FIRST_NAME) != null && attrs.get(USER_FIRST_NAME).getValue() != null) {
       String firstName = attrs.get(USER_FIRST_NAME).getValue().toString();
       changed |= checkIfChanged && !Objects.equals(firstName, user.getFirstName());
       user.setFirstName(firstName);
@@ -127,29 +127,29 @@ public class EntityMapperUtils {
       }
       changed |= checkIfChanged && !Objects.equals(originalLastLoginDate, lastLoginDate);
     }
-    if (attrs.containsKey(USER_LAST_NAME)) {
+    if (attrs.containsKey(USER_LAST_NAME) && attrs.get(USER_LAST_NAME) != null && attrs.get(USER_LAST_NAME).getValue() != null) {
       String lastName = attrs.get(USER_LAST_NAME).getValue().toString();
       changed |= checkIfChanged && !Objects.equals(lastName, user.getLastName());
       user.setLastName(lastName);
     }
-    if (attrs.containsKey(USER_DISPLAY_NAME)) {
+    if (attrs.containsKey(USER_DISPLAY_NAME) && attrs.get(USER_DISPLAY_NAME) != null && attrs.get(USER_DISPLAY_NAME).getValue() != null) {
       // TODO: GTNPORTAL-2358 Change once displayName will be available as
       // part of Organization API
       String fullName = attrs.get(USER_DISPLAY_NAME).getValue().toString();
       changed |= checkIfChanged && !Objects.equals(fullName, user.getDisplayName());
       user.setDisplayName(fullName);
     }
-    if (attrs.containsKey(USER_ORGANIZATION_ID)) {
+    if (attrs.containsKey(USER_ORGANIZATION_ID) && attrs.get(USER_ORGANIZATION_ID) != null && attrs.get(USER_ORGANIZATION_ID).getValue() != null) {
       String organizationId = attrs.get(USER_ORGANIZATION_ID).getValue().toString();
       changed |= checkIfChanged && !Objects.equals(organizationId, user.getOrganizationId());
       user.setOrganizationId(organizationId);
     }
-    if (attrs.containsKey(USER_CREATION_SOURCE)) {
+    if (attrs.containsKey(USER_CREATION_SOURCE) && attrs.get(USER_CREATION_SOURCE) != null && attrs.get(USER_CREATION_SOURCE).getValue() != null) {
       String creationSource = attrs.get(USER_CREATION_SOURCE).getValue().toString();
       changed |= checkIfChanged && !Objects.equals(creationSource, user.getCreationSource());
       user.setCreationSource(creationSource);
     }
-    if (attrs.containsKey(USER_ENABLED)) {
+    if (attrs.containsKey(USER_ENABLED) && attrs.get(USER_ENABLED) != null && attrs.get(USER_ENABLED).getValue() != null) {
       // used when populating User from AD ; it returns numbers : 512 = enbaled, 514 = disabled
       String status = attrs.get(USER_ENABLED).getValue().toString();
 
@@ -170,14 +170,14 @@ public class EntityMapperUtils {
         ((UserImpl) user).setEnabled(enabled);
       }
 
-      if (attrs.containsKey(AUTOMATIC_DISABLE)) {
+      if (attrs.containsKey(AUTOMATIC_DISABLE) && attrs.get(AUTOMATIC_DISABLE).getValue() != null && attrs.get(AUTOMATIC_DISABLE).getValue().toString() != null) {
         String automaticDisabledString = attrs.get(AUTOMATIC_DISABLE).getValue().toString();
         boolean automaticDeactivation = StringUtils.equals(automaticDisabledString, "true");
         changed |= checkIfChanged && !Objects.equals(automaticDeactivation, user.isAutomaticDeactivation());
         user.setAutomaticDeactivation(automaticDeactivation);
       }
     }
-    if (user instanceof UserImpl && attrs.containsKey(ORIGINATING_STORE)) {
+    if (user instanceof UserImpl && attrs.containsKey(ORIGINATING_STORE) && attrs.get(ORIGINATING_STORE).getValue() != null && attrs.get(ORIGINATING_STORE).getValue().toString() != null) {
       UserImpl userImpl = (UserImpl) user;
       userImpl.setOriginatingStore(attrs.get(ORIGINATING_STORE).getValue().toString());
     }

